gnomad_hail/utils/sample_qc.py

```diff
@@ -278,85 +278,69 @@ def assign_platform_from_pcs(
-    sex_ht = sex_ht.annotate(
-        is_female=(
-            hl.case()
-                .when(sex_ht.no_y_call_rate_platforms.contains(sex_ht.qc_platform), sex_ht.is_female)
-                .when(sex_ht.is_female & ((sex_ht.y_call_rate - y_female_stats.min) / (y_male_stats.max - y_female_stats.min) < max_y_female_call_rate), True)
-                .when(~sex_ht.is_female & ((sex_ht.y_call_rate - y_female_stats.min) / (y_male_stats.max - y_female_stats.min) > min_y_male_call_rate), False)
-                .or_missing()
-        )
-    )
-    sex_ht = sex_ht.annotate_globals(
-        impute_sex_params=hl.struct(
-            male_min_f_stat=male_min_f_stat,
-            female_max_f_stat=female_max_f_stat,
-            min_male_y_sites_called=min_male_y_sites_called,
-            max_y_female_call_rate=max_y_female_call_rate,
-            min_y_male_call_rate=min_y_male_call_rate
+    Imputes sample sex based on X-chromosome heterozygosity and sex chromosome ploidy. 
+    Returns Table with the following fields:
+    - s (str): Sample
+    - f_stat (float64): Sample f-stat. Calculated using hl.impute_sex.
+    - n_called (int64): Number of variants with a genotype call. Calculated using hl.impute_sex.
+    - expected_homs (float64): Expected number of homozygotes. Calculated using hl.impute_sex.
+    - observed_homs (int64): Expected number of homozygotes. Calculated using hl.impute_sex.
+    - X_karyotype (str): Sample's chromosome X karyotype.
+    - Y_karyotype (str): Sample's chromosome Y karyotype.
+    - sex_karyotype (str): Sample's sex karyotype.
+
+    :param mt: Input MatrixTable
+    :param ploidy_ht: Table of samples with their imputed sex chromosome ploidies
+    :param sites_ht: Optional Table to use. If present, filters input MatrixTable to sites in this Table prior to imputing sex.
+    :param aaf_expr: Optional. Name of field in input MatrixTable with alternate allele frequency.
+    :param gt_expr: Name of entry field storing the genotype. Default: 'GT'
+    :param f_stat_cutoff: f-stat to roughly divide 'XX' from 'XY' samples. Assumes XX samples are below cutoff and XY are above cutoff.
+    :param float aaf_threshold: Minimum alternate allele frequency to be used in f-stat calculations.
+    :return: Table of samples and their imputed sex karyotypes.
+    """
+    x_contigs = get_reference_genome(mt.locus).x_contigs
+    logger.info(f"Filtering mt to biallelic SNPs in X contigs: {x_contigs}")
+    if 'was_split' in list(mt.row):
+        mt = mt.filter_rows((~mt.was_split) & hl.is_snp(mt.alleles[0], mt.alleles[1]))
+    else:
+        mt = mt.filter_rows((hl.len(mt.alleles) == 2) & hl.is_snp(mt.alleles[0], mt.alleles[1]))
+    mt = hl.filter_intervals(mt, [hl.parse_locus_interval(contig) for contig in x_contigs])
+
+    if sites_ht:
+        if aaf_expr == None:
+            logger.warn("sites_ht was provided, but aaf_expr is missing. Assuming name of field with alternate allele frequency is 'AF'.")
```

Comment:
The docstring doesn't mention that the sites HT is used for AF purpose; add to the docstring?
Also, I'm wondering about the benefit of having `sites_ht` as an argument at all; seems more intuitive for users to filter/annotate to the sites they want outside of the function. Now if bringing the ploidy computation in there may be more of a case to be made as you'd want  all sites for that.